### PR TITLE
Access application routes from the application context

### PR DESCRIPTION
**Pull request by waiting-for-dev**
Relates to https://github.com/hanami/hanami/pull/1119

### CODE REVIEW

File: spec/integration/context/application_context/routes_spec.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            class Context < Hanami::View::Context
+            end
+          end
+        end
+      RUBY
+
+      require "hanami/init"
+
+      context = TestApp::View::Context.new
+      expect(context.routes.path(:root)).to eq "/"
```
Comment (waiting-for-dev) [inline spec/integration/context/application_context/routes_spec.rb:42]: @timriley, for some reason, this test is flaky. It seems that sometimes` Hanami.application[:routes_helper]` is not defined, so `context.routes` ends up being `nil`. Do you have an idea what could be going on?

Comment (timriley) [reply spec/integration/context/application_context/routes_spec.rb:42]: @waiting-for-dev Definitely happy to take a look. Do you have an RSpec command and a seed I can use to replicate this, by any chance?

Comment (waiting-for-dev) [reply spec/integration/context/application_context/routes_spec.rb:42]: Thanks, @timriley. Sure! Here it goes:

```bash
bundle exec rspec --seed=26933 spec/integration/context/application_context/routes_spec.rb
```

I thought that maybe something would come up to your mind quickly, but if not, I'll also look further into it :slightly_smiling_face:

Comment (waiting-for-dev) [reply spec/integration/context/application_context/routes_spec.rb:42]: If you run tests for that single file, the error only happens when the problematic test is run in the second place. Probably, it doesn't work when another integration test has been run before. I'd say that we're missing something in the teardown step. I tried with some tweaks, but no success so far. It seems related to Hanami itself and not Zeitwek. If I rename `TestApp` in one of the two tests, the problem remains.

It's the only place where we're using `with_tmp_directory` if that helps.

Comment (timriley) [reply spec/integration/context/application_context/routes_spec.rb:42]: Fixed in https://github.com/hanami/view/pull/197/commits/6f848d8ab175b09ced82e06443c728fb4861180b 😄

**Comment by timriley** [comment]
@waiting-for-dev Fixed that test issue for you! Is this PR otherwise ready to go, in your eyes? (Just noting that it's still marked as draft)

**Comment by waiting-for-dev** [comment]
> @waiting-for-dev Fixed that test issue for you! Is this PR otherwise ready to go, in your eyes? (Just noting that it's still marked as draft)

Thanks, @timriley! :heart: :heart: Yeah, it's ready to go, it was a draft just because the flaky test :slightly_smiling_face:

**Comment by timriley** [comment]
Thanks @waiting-for-dev 😄

**Pull request merged**